plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/convert/ConvertProcessor.java

```diff
@@ -129,12 +132,12 @@ public String getType() {
     public static class Factory implements Processor.Factory<ConvertProcessor> {
         @Override
         public ConvertProcessor create(Map<String, Object> config) throws Exception {
-            Map<String, String> fields = ConfigurationUtils.readMap(config, "fields");
-            Map<String, Type> convertFields = new HashMap<>();
-            for (Map.Entry<String, String> entry : fields.entrySet()) {
-                convertFields.put(entry.getKey(), Type.fromString(entry.getValue()));
+            if (config.size() != 1) {
+                throw new IllegalArgumentException("The processor only renames one field at a time.");
```

Comment:
this message might get confusing if people don't specify any fields, btw we should add a test for that.